rama-core/src/extensions.rs

```diff
@@ -27,206 +27,88 @@
 //! assert_eq!(ext.get::<i32>(), Some(&5i32));
 //! ```
 
-use rama_utils::macros::generate_set_and_with;
 use std::any::{Any, TypeId};
-use std::collections::HashMap;
 use std::fmt;
-use std::hash::{BuildHasherDefault, Hasher};
 use std::ops::{Deref, DerefMut};
 use std::pin::Pin;
 use std::sync::Arc;
 
-#[cfg(debug_assertions)]
-use std::any::type_name;
-
-type AnyMap = HashMap<TypeId, Box<dyn AnyClone + Send + Sync>, BuildHasherDefault<IdHasher>>;
-#[cfg(debug_assertions)]
-type TypeIdMap = HashMap<TypeId, String, BuildHasherDefault<IdHasher>>;
-
-// With TypeIds as keys, there's no need to hash them. They are already hashes
-// themselves, coming from the compiler. The IdHasher just holds the u64 of
-// the TypeId, and then returns it, instead of doing any bit fiddling.
-#[derive(Default)]
-struct IdHasher(u64);
-
-impl Hasher for IdHasher {
-    fn write(&mut self, _: &[u8]) {
-        unreachable!("TypeId calls write_u64");
-    }
-
-    #[inline(always)]
-    fn write_u64(&mut self, id: u64) {
-        self.0 = id;
-    }
-
-    #[inline(always)]
-    fn finish(&self) -> u64 {
-        self.0
-    }
-}
-
 /// A type map of protocol extensions.
 ///
 /// `Extensions` can be used by `Request` and `Response` to store
 /// extra data derived from the underlying protocol.
 #[derive(Clone, Default)]
 pub struct Extensions {
-    // If extensions are never used, no need to carry around an empty HashMap.
-    // That's 3 words. Instead, this is only 1 word.
-    map: Option<Box<AnyMap>>,
-    #[cfg(debug_assertions)]
-    type_map: Option<Box<TypeIdMap>>,
-    parent_extensions: Option<Arc<Extensions>>,
+    // TODO optimize this storage
```

Comment:
What do you want to optimize? Would help to expanda bit more on these kind of comments. E.g. the problem statement, the goal, the ideas, ...